docs/advanced-usage.rst

```diff
@@ -127,15 +127,46 @@ You can connect to a proxy using HTTP, HTTPS or SOCKS. urllib3's behavior will
 be different depending on the type of proxy you selected and the destination
 you're contacting.
 
-When contacting a HTTP website through a HTTP or HTTPS proxy, the request will
-be forwarded with the `absolute URI
-<https://tools.ietf.org/html/rfc7230#section-5.3.2>`_.
+HTTP and HTTPS proxies
+~~~~~~~~~~~~~~~~~~~~~~
+
+Both HTTP/HTTPS proxies support HTTP and HTTPS destinations. The only
+difference between them is if you need to establish a TLS connection to the
+proxy first. You can specify which proxy you need to contact by specifying the
+proper proxy scheme. (i.e ``http://`` or ``https://``)
+
+urllib3's behavior will be different depending on your proxy and destination:
+
+* *HTTP proxy - HTTP destination*
+   Your request will be forwarded with the `absolute URI
+   <https://tools.ietf.org/html/rfc7230#section-5.3.2>`_.
+
+* *HTTP proxy - HTTPS destination*
+    A TCP tunnel will be established with a HTTP CONNECT. Afterward a TLS
+    connection will be established with the destination and your request will
+    be sent.
+
+* *HTTPS proxy - HTTP destination*
+    A TLS connection will be established to the proxy and later your request
+    will be forwarded with the `absolute URI
+    <https://tools.ietf.org/html/rfc7230#section-5.3.2>`_.
+
+* *HTTPS proxy - HTTPS destination*
+    A TLS-in-TLS tunnel will be established.  The first TLS connection will be
+    to the proxy, the second one to the destination. You can customize the
+    ``SSLContext`` used for the proxy TLS connection through the
+    ``proxy_ssl_context`` argument of the :class:`~poolmanager.ProxyManager`
+    class.
+
+For HTTPS proxies we also support forwarding your requests to HTTPS destinations with
+an `absolute URI <https://tools.ietf.org/html/rfc7230#section-5.3.2>`_ if the
+``use_forwarding_for_https`` argument is set. We **strongly recommend** you
```

Comment:
My first reading of this had me thinking we strongly recommend "using this option". Maybe we should take the emphasis off strongly recommend and put it on "only use this option with trusted or corporate proxies"

also nit: "argument is set to `True`"